### PR TITLE
squirrel-sql: Add jtds as a driver.

### PR DESCRIPTION
**Pull request by cresh**
###### Motivation for this change

Add jTDS driver for squirrel-sql for connecting to MSSQL databases.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/mssql/jdbc/jtds.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{lib, stdenv, fetchurl, unzip}:
+
+stdenv.mkDerivation rec {
+  pname = "jtds";
+  version = "1.3.1";
+
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/jtds/files/jtds/${version}/${pname}-${version}-dist.zip";
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/mssql/jdbc/jtds.nix:8]: Please use a mirror:// style URL.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/sql/mssql/jdbc/jtds.nix (hunk 1 above)
Comment (cresh) [reply pkgs/servers/sql/mssql/jdbc/jtds.nix]: Thanks for the review, I've changed my pull request to use a mirror URL.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -21248,6 +21248,7 @@ with pkgs;
   mysql_jdbc = callPackage ../servers/sql/mysql/jdbc { };
 
   mssql_jdbc = callPackage ../servers/sql/mssql/jdbc { };
+  jtds_jdbc = callPackage ../servers/sql/mssql/jdbc/jtds.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21251]:
```suggestion
  jtds_jdbc = callPackage ../servers/sql/mssql/jdbc { };
```

File: pkgs/servers/sql/mssql/jdbc/jtds.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    cp jtds-*.jar $out/share/java/jtds-jdbc.jar
+  '';
+
+  nativeBuildInputs = [ unzip ];
+
+  meta = with lib; {
+    description = "Pure Java (type 4) JDBC 3.0 driver for Microsoft SQL Server";
+    homepage = "http://jtds.sourceforge.net/";
+    license = licenses.lgpl21;
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/mssql/jdbc/jtds.nix:25]:
```suggestion
    platforms = platforms.unix;
```
not required with line 21.

Comment (cresh) [reply pkgs/servers/sql/mssql/jdbc/jtds.nix]: Thanks!

**Pull request merged**